opentelemetry-sdk/tests/trace/propagation/test_jaeger_format.py

```diff
@@ -0,0 +1,120 @@
+        ),
+        parent=parent.get_span_context(),
+    )
+
+    new_carrier = {}
+    ctx = trace_api.set_span_in_context(child, ctx)
+
+    FORMAT.inject(dict.__setitem__, new_carrier, context=ctx)
+
+    return ctx, new_carrier
+
+
+class TestJaegerFormat(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        ids_generator = trace_api.RandomIdsGenerator()
+        cls.trace_id = ids_generator.generate_trace_id()
+        cls.span_id = ids_generator.generate_span_id()
+        cls.parent_span_id = ids_generator.generate_span_id()
+        cls.serialized_uber_trace_id = jaeger_format.format_uber_trace_id(
+            cls.trace_id, cls.span_id, cls.parent_span_id, 1
+        )
+
+    def test_extract_valid_span(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        ctx = FORMAT.extract(carrier_getter, old_carrier)
+        span_context = trace_api.get_current_span(ctx).get_span_context()
+        self.assertEqual(span_context.trace_id, self.trace_id)
+        self.assertEqual(span_context.span_id, self.span_id)
+
+    def test_trace_id(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        _, new_carrier = get_context_new_carrier(old_carrier)
+        self.assertEqual(
+            self.serialized_uber_trace_id.split(":")[0],
+            new_carrier[FORMAT.TRACE_ID_KEY].split(":")[0],
+        )
+
+    def test_parent_span_id(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        _, new_carrier = get_context_new_carrier(old_carrier)
+        span_id = self.serialized_uber_trace_id.split(":")[1]
+        parent_span_id = new_carrier[FORMAT.TRACE_ID_KEY].split(":")[2]
+        self.assertEqual(span_id, parent_span_id)
+
+    def test_sampled_flag(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        _, new_carrier = get_context_new_carrier(old_carrier)
+        sample_flag_value = (
+            int(new_carrier[FORMAT.TRACE_ID_KEY].split(":")[3]) & 0x01
+        )
+        self.assertEqual(1, sample_flag_value)
+
+    def test_baggage(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        input_baggage = {"key1": "value1"}
+        _, new_carrier = get_context_new_carrier(old_carrier, input_baggage)
+        ctx = FORMAT.extract(carrier_getter, new_carrier)
+        self.assertDictEqual(input_baggage, ctx["baggage"])
+
+    def test_non_string_baggage(self):
+        old_carrier = {FORMAT.TRACE_ID_KEY: self.serialized_uber_trace_id}
+        input_baggage = {"key1": 1, "key2": True}
+        formatted_baggage = {"key1": "1", "key2": "True"}
+        _, new_carrier = get_context_new_carrier(old_carrier, input_baggage)
+        ctx = FORMAT.extract(carrier_getter, new_carrier)
+        self.assertDictEqual(formatted_baggage, ctx["baggage"])
+
+    def test_extract_invalid_uber_trace_id(self):
```

Comment:
would be good to add a unit test to verify the behavior of valid baggage, invalid trace key, and visa versa.

This would help document expected behavior as well.